### DISCUSSION TITLE
Classification - execution taking longer than normal

### DISCUSSION THREAD

**Discussion by vahahn** [pycaret/pycaret#3441] [General]
Hello,

I have a fairly small-ish dataset (~6000 rows, 7 cols). I ran a script that I've been running for the past 3 months, but this time it's taking much longer than normal (used to take a couple of minutes). I have approx 1000 more rows of data than I did previously and I've read from a few posts that many are working with much larger datasets, so I'm assuming that I may be doing something wrong. 

If anyone can shed some light then that would be warmly appreciated!

This is what I am currently reading at the bottom
`Executing (1h 1m 36s) Cell navigate_next setup() navigate_next setup() navigate_next _prepare_dataset() navigate_next merge() navigate_next merge() navigate_next get_result() navigate_next _get_join_info() navigate_next join() navigate_next join() navigate_next _join_non_unique()`

**_Here's the script:_**

```
from pycaret.classification import *

clf1 = setup(cta_call, 
             target = 'cta_call', 
             log_experiment=True, 
             experiment_name='compare_models')
             feature_selection = True, 
             remove_multicollinearity = True)

best_model = compare_models()
print(best_model)
evaluate_model(best_model)
```

pycaret version: 3.0.0
google collab: 3.9.16

**Comment by ngupta23** [comment]
Lets track it here: https://github.com/pycaret/pycaret/issues/3442
